main.cpp

```diff
@@ -0,0 +1,228 @@
+using std::string;
+using std::cin;
+
+const string CONFIG_FILE_PATH = "/users/bconway/SourceTree/School/CPT-180-27-Assignment-2-Config-Generator/app.cfg";
+
+
+int main(int argc, char *argv[]) {
+
+    string argument1 = "";
+    // Make sure that the argument isn't empty
+    if (argv[1] != nullptr) {
+        argument1 = static_cast<string>(argv[1]);
+    }
+
+    if (argument1 == "init") {
+
+        // initialize the config file
+        // TODO determine if the file exists and handle the exception
+
+        std::ofstream config_file;
+
+        string temp_value = "";
+
+        config_file.open(CONFIG_FILE_PATH);
+
+        cout << "Enter your name (first and last):" << endl;
+        std::getline(cin, temp_value);
+
+        // If nothing was entered then ask again.
+        while (temp_value == "") {
+            cout << "Error, please enter your name (first and last):" << endl;
+            std::getline(cin, temp_value);
+        }
+        config_file << "[name]=" << temp_value << endl;
+        temp_value = "";
+
+        cout << "Enter your email address:" << endl; // TODO validate email address
+        std::getline(cin, temp_value);
+
+        // If nothing was entered then ask again.
+        while (temp_value == "") {
+            cout << "Error, please enter your email address:" << endl; // TODO validate email address
+            std::getline(cin, temp_value);
+        }
+        config_file << "[email]=" << temp_value << endl;
+        temp_value = "";
+
+        cout << "Enter your unique cypher:" << endl;
+        std::getline(cin, temp_value);
+
+        // If nothing was entered then ask again.
+        while (temp_value == "") {
+            cout << "Error, please enter your unique cypher:" << endl;
+            std::getline(cin, temp_value);
+        }
+        config_file << "[cypher]=" << temp_value << endl;
+        temp_value = "";
+
+        cout << "Enter your timezone-offset:" << endl;
+        std::getline(cin, temp_value);
+
+        // If nothing was entered then ask again.
+        while (temp_value == "") {
+            cout << "Error, please enter your timezone-offset:" << endl;
+            std::getline(cin, temp_value);
+        }
+        config_file << "[timezone]=" << temp_value << endl;
+        temp_value = "";
+
+        cout << "Enter path to knownrecipients file:" << endl;
+        std::getline(cin, temp_value);
+
+        // Set a default value if nothing was entered.
+        if (temp_value == "") {
+            config_file << "[knownrecipients]=./knownrecipients.txt" << endl;
+        } else {
+            config_file << "[knownrecipients]=" << temp_value << endl;
+        }
+
+        temp_value = "";
+
+        config_file.close();
+
+
+    } else if (argument1 == "edit") {
+        // edit the config file
+
+        // Create a backup of the config file
+        const char * orig_file = CONFIG_FILE_PATH.c_str();
+        const char * backup_file = (CONFIG_FILE_PATH + ".old").c_str();
```

Comment:
I am going to try it myself right now. It seems to work with string literals. Might just be something silly about C.